### PR TITLE
Require Open Omi and PTT shortcuts during onboarding

### PR DESCRIPTION
**Pull request by Git-on-my-level**
## What changed and why

Require every new macOS user to choose and exercise both the Open Omi and push-to-talk shortcuts before onboarding can continue. The shortcut steps now offer an explicit Custom shortcut choice, keep the full-onboarding skip unavailable until both required stages are complete, and reuse the existing persisted shortcut settings. PTT custom recording now rejects bare keys everywhere because a system-wide bare letter would start a voice turn during ordinary typing.

The onboarding shell is centered on first launch, and Back/Skip live in the onboarding card header instead of floating outside the card.

## Product invariants affected

- INV-CHAT-1

## How it was verified

- `./scripts/dev-feedback.py --once swift 'SBOnboardingBackNavigationTests|SBOnboardingShortcutRecordingTests|ShortcutSettingsTests|ShellSummonTests|GlobalShortcutManagerRegistrationTests'` — 68 tests passed.
- `./scripts/swift-format-wrapper.sh lint -r <changed Swift files>` — passed.
- `python3 scripts/check_desktop_test_quality.py` — passed.
- Built and launched the isolated signed `/Applications/omi-shortcut-onboarding.app` bundle from this branch with the development backend and automation bridge on port 47941. Relaunched in normal UI mode, reset onboarding, and visually confirmed the shell/card are centered and Back is inside the card header.
- In the real onboarding UI, confirmed both shortcut stages omit Skip, Custom shortcut is available, Continue remains absent after selection/recording, and appears only after the chosen shortcut is pressed. Exercised Open Omi with `⌃⌘O`; pressed bare `u` in custom PTT recording and confirmed the inline refusal; the offered modifier-only PTT choices remain available.

## Tests

`SBOnboardingBackNavigationTests.testShortcutStagesCannotAdvanceUntilSelectedShortcutIsExercised` covers the required selection/exercise gate. `testFullOnboardingSkipUnlocksOnlyAfterRequiredShortcutStages` covers the onboarding skip boundary, `ShortcutSettingsTests.testPushToTalkRequiresAtLeastOneModifier` covers the shared PTT policy, and `GlobalShortcutManagerRegistrationTests.testOnboardingValidationSurfacesAConflictingAskOmiChord` covers the Carbon registration check before Open Omi can advance.

## Failure class (fixes)

Failure-Class: none

<a href="https://cubic.dev/pr/BasedHardware/omi/pull/11475?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Comment by Git-on-my-level** [comment]
Acknowledged. The arm/disarm pattern intentionally suspends Carbon registration during the shortcut step so the local monitor can observe the keystroke, and real registration is deferred to `disarmShortcutSummon()`. A conflicting chord could pass the onboarding test press but fail the deferred Carbon `registerAskOmi()` — that's a real edge case.

This PR adds the requirement gate (select + exercise) to the existing onboarding shortcut flow; the arm/disarm design and its registration-deferral semantics predate it. Adding a Carbon registration probe or conflict-surfacing UI during onboarding is a separate feature with its own API surface (`GlobalShortcutManager.registerAskOmi` outcome handling + new onboarding failure state). Tracking it separately rather than expanding this PR's scope.

The default chords (⌃⌘O for Open Omi, fn/opt/ctrl for PTT) are very low conflict risk — ⌃⌘O is already the unconditionally-registered summon chord.

File: desktop/macos/Desktop/Sources/Onboarding/SecondBrain/SBOnboardingModel+Steps.swift [hunk 1]
```diff
@@ -702,10 +707,13 @@ extension SBOnboardingModel {
   }
 
   func answerShortcutOpen() {
-    advance(userAnswer: shortcutPressed ? "Works" : (shortcutPicked ? "Set" : "Skip"), to: .shortcutTalk)
+    guard shortcutPicked, shortcutPressed else { return }
+    advance(userAnswer: "Works", to: .shortcutTalk)
```
[the remark this answers is not in the document]
Comment (Git-on-my-level) [reply desktop/macos/Desktop/Sources/Onboarding/SecondBrain/SBOnboardingModel+Steps.swift:721]: Addressed in aeb02ade2b. Before the Open Omi stage advances, onboarding now performs a temporary Carbon registration probe while the local monitor is suspended; conflicts keep the stage active with an inline error. The probe is removed and suspension is restored after validation. Added GlobalShortcutManagerRegistrationTests.testOnboardingValidationSurfacesAConflictingAskOmiChord, and exercised the centered QA bundle through the Open Omi test step.
